### PR TITLE
adding 'sitemap generator' page under 'gems'

### PR DESCRIPTION
**Pull request by oliveranthony17**
I have added the sitemap stuff under 'gems' section - but this also includes instructions on the bin script and github action. Let me know if you think there's a better place to put this.

@ignaciosy I have requested review from you given we worked on this together - a quick proof read would be great 👍

### CODE REVIEW

**Review by ignaciosy — APPROVED** [review]
💯 awesome! Great contribution

**Review by coorasse — COMMENTED** [review]
I disagree with this approach: I think the sitemap should not belong to the project source code.
I think a correct example is how it is done in schuler: we have a night scheduler that generates the sitemap in the running app and saves the file, pings search engine, and can provide different sitemaps for different envs.

File: ruby_on_rails/sitemap_generator.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+Here you will learn how to implement a Sitemap Generator into your project. This guide also includes details on how you can create a script that automatically updates the sitemap file, and a GitHub Action that will automatically run the script and push the changes to a new pull request.
+
+## Implementation
+
+For implementing the sitemap generator itself, you can check out the gem repository [here](https://github.com/kjvarga/sitemap_generator). It contains all the necessary information on how to initialize the generator and how to configure the sitemap. Pay attention to the [alternate links section](https://github.com/kjvarga/sitemap_generator#alternate-links) which provides details on how to add alternate links to the sitemap where there are multiple locales available.
+
+In the example below, we have also added `nil` to the locales, in order to generate the sitemap for all the available locales, but also with no locale specified. This can be used as inspiration, but the requirements will differ for every project.
+
+```ruby
```
Comment (schmijos) [inline ruby_on_rails/sitemap_generator.md:11]: Please metion the file name with a comment.

**Comment by schmijos** [comment]
> I disagree with this approach: I think the sitemap should not belong to the project source code. I think a correct example is how it is done in schuler: we have a night scheduler that generates the sitemap in the running app and saves the file, pings search engine, and can provide different sitemaps for different envs. Also no deployment is needed when the sitemap changes.

I don't get your point. It's a rake task per this PR already, right?

**Comment by coorasse** [comment]
This part: https://github.com/renuo/applications-setup-guide/pull/345/files#diff-16ff326e018d9fcfaec0c347eb61fb8da0d82fd49c8f0e41772c7e35b73d786dR109-R118

**Comment by coorasse** [comment]
What's the status here?

**Comment by hunchr** [comment]
@oliveranthony17 ^

**Comment by coorasse** [comment]
If there's no interest in pushing this forward, I would close it, because I don't think is the right approach. I am available to discuss :)

**Comment by oliveranthony17** [comment]
OK I discussed with @coorasse and we will change our approach @ignaciosy (the sitemap saga continues... 🤣). We can use an S3 bucket and rake task on Heroku to automatically produce the sitemap.

I will change this on the website, and also here when I get some time... Therefore I will leave open as draft for now.

**Pull request closed without merging**